src/ui/public/field_editor/field_editor.js

```diff
@@ -129,6 +154,14 @@ uiModules
         else return fieldFormats.getDefaultType(self.field.type);
       }
 
+      function getScriptingLangs() {
+        return $http.get(chrome.addBasePath('/api/kibana/scripts/languages'))
+        .then((res) => res.data)
+        .catch(() => {
+          return $q.reject(new Error('Error getting available scripting languages from Elasticsearch'));
```

Comment:
Not sure how consistently Kibana uses these, but `ui/public/errors` has a lot of typed errors. You could reuse `RequestFailure` here.